app/src/main/java/com/realityexpander/tasky/MainActivity.kt

```diff
@@ -46,80 +46,58 @@ class MainActivity : ComponentActivity() {
-                        LaunchedEffect(true) {
-                            val appSettings = context.dataStore.data.first()
-
-                            // Confirm the settings file is created and initialized
-                            if (!appSettings.isSettingsInitialized) {
-                                context.dataStore.saveSettingsInitialized(true)
-                            }
-
-                            // Set user logged-in status
-                            viewModel.onSetAuthInfo(appSettings.authInfo)
+        setContent {
+            TaskyTheme {
+                Surface(
+                    modifier = Modifier.fillMaxSize(),
+                    color = colorResource(id = R.color.tasky_green)
+                ) {
+                    val splashState by viewModel.splashState.collectAsState()
+                    val context = LocalContext.current
+
+                    val navController = rememberNavController()
+                    val navHostEngine = rememberNavHostEngine()
+
+                    // Load Settings (or initialize them)
+                    LaunchedEffect(true) {
+                        val appSettings = context.dataStore.data.first()
+
+                        // Confirm the settings file is created and initialized
+                        if (!appSettings.isSettingsInitialized) {
+                            context.dataStore.saveSettingsInitialized(true)
                         }
 
-                        if (!splashState.isLoading) {
+                        // Set logged-in user Authentication status
+                        viewModel.onSetAuthInfo(appSettings.authInfo)
+                    }
 
-                            // Check for errors
-                            if (splashState.error != null) {
-                                Toast.makeText(context, splashState.error, Toast.LENGTH_LONG).show()
-                                Thread.sleep(2000)
-                                viewModel.onSetAuthInfo(null)
-                            }
+                    if (!splashState.isLoading) {
 
-                            DestinationsNavHost(
-                                navGraph = NavGraphs.root,
-                                navController = navController,
-                                engine = navHostEngine,
-                                startRoute =
-                                if (splashState.authInfo != null) {
-                                    AgendaScreenDestination
-                                } else {
-                                    LoginScreenDestination
-                                },
-                            )
+                        // Check for errors
+                        if (splashState.error != null) {
+                            Toast.makeText(context, splashState.error, Toast.LENGTH_LONG).show()
+                            Thread.sleep(1000)
+                            viewModel.onSetAuthInfo(null)
                         }
```

Comment:
This should go in a `LaunchedEffect` as well, since these functions aren't composables

Reply:
## FOLLOW UP QUESTION 2

Is the reason for putting these in a Launch block for performance? Or is there another reason to put these non-composable functions into a `Launchedeffect` block?

Reply:
It's just that composables should be side-effect free. In this particular case it probably won't cause any harm, but in general composable functions can execute uncontrolled in any order which is why you want to avoid placing non-compose code in them. LaunchedEffect takes care of exactly that by providing a scope that is cancelled and relaunched when the given state changes